sound/soc/sof/intel/hda-codec.c

```diff
@@ -142,9 +142,9 @@ static int hda_codec_probe(struct snd_sof_dev *sdev, int address,
 	if (!hda_priv)
 		return -ENOMEM;
 
-	hda_priv->codec.bus = hbus;
-	hdev = &hda_priv->codec.core;
-	codec = &hda_priv->codec;
+	hda_priv->codec->bus = hbus;
```

Comment:
how does this work?

hda_priv is initialized with zeroes, so hda_priv->codec is NULL.

Reply:
wouldn't this break git bisect?

Reply:
if we can't use git bisect to run tests, that's a deal breaker. Sorry.

Reply:
Will see what I can do. Perhaps I'll define the new functions first, then do field -> pointer + skl & sof update in single following patch.

Reply:
Thanks. git bisect is heavily used to root-cause upstream regressions or side effects of changes in programming sequences, I always insist that we keep the functionality even when doing open-heart surgery in the code.